stdlib/array.gr

```diff
@@ -892,3 +895,49 @@ export let sort = (comp, array) => {
   let len = length(array)
   quicksort(0, len - 1)
 }
+
+/**
+ * Rotates an array by n elements to the right, in place.
+ *
+ * If n is negative, the array will be rotated by n elements
+ * to the left. See examples.
+ *
+ * @param arr: The array to be rotated
+ * @param n: The number of elements to rotate by
+ * @returns The rotated array
+ *
+ * @example: rotate([> 1, 2, 3, 4, 5], 2) == [> 4, 5, 1, 2, 3]
+ * @example: rotate([> 1, 2, 3, 4, 5], -1) == [> 2, 3, 4, 5, 1]
+ * @since v0.4.5
+ */
+export let rotate = (arr, n) => {
+  let rec gcd = (a, b) => {
+    if (b == 0) {
+      a
+    }
+    else{
+      gcd(b, a % b)
+    }
+  }
+
+  let arrLen = length(arr)
+  let k = n % arrLen
+  let mut d = -1
+  let mut temp = 0
+  let mut j  = 0
+  for (let mut i = 0; i < gcd(arrLen, k); i += 1) {
+    j = i
+    temp = arr[i]
+    while (true) {
+      d = (j - k) % arrLen
+      if (d == i) {
+        break
+      }
+      let newVal = arr[d]
+      arr[j] = newVal
+      j = d
+    }
+    arr[j] = temp
+  }
+  arr
```

Comment:
I believe we are going to remove the return value since this is a mutating function
```suggestion
```